theano/gpuarray/corr_gemm.c

```diff
@@ -429,21 +444,56 @@ PyGpuArrayObject* corrMM(PyGpuArrayObject *const bottom,
     const size_t topHeight = _CONV_FLOORDIV_X(topHeightNoDH, dH) + 1;
     const size_t topWidth  = _CONV_FLOORDIV_X(topWidthNoDW, dW) + 1;
 #undef _CONV_FLOORDIV
-    if (batchSize != PyGpuArray_DIMS(top)[0] ||
-            nFilters != PyGpuArray_DIMS(top)[1] ||
-            topHeight != PyGpuArray_DIMS(top)[2] ||
-            topWidth != PyGpuArray_DIMS(top)[3]) {
-        PyErr_Format(PyExc_ValueError,
-                "GpuCorrMM shape inconsistency:\n"
-                "  bottom shape: %ld %ld %ld %ld\n"
-                "  weight shape: %ld %ld %ld %ld\n"
-                "  top shape: %ld %ld %ld %ld (expected %ld %ld %ld %ld)\n",
-                batchSize, nChannels, bottomHeight, bottomWidth,
-                nFilters, nChannels, kH, kW,
-                PyGpuArray_DIMS(top)[0], PyGpuArray_DIMS(top)[1],
-                PyGpuArray_DIMS(top)[2], PyGpuArray_DIMS(top)[3],
-                batchSize, nFilters, topHeight, topWidth);
-        return NULL;
+    if (unshared) {
+        if (topHeight != PyGpuArray_DIMS(weight)[1] ||
+                topWidth != PyGpuArray_DIMS(weight)[2]) {
+            PyErr_Format(PyExc_ValueError,
+                    "GpuCorrMM regions in kernel must match output regions:\n"
+                    "  bottom shape: %d %d %d %d\n"
+                    "  weight shape: %d %ld %ld %d %d %d"
+                    " (expected %d %d %d %d %d %d)\n"
+                    "  top shape(calculated): %d %d %d %d\n",
```

Comment:
Since the shared version uses `%ld` everywhere, it's probably good to do that here as well. (My compiler gave some warnings here.)